consensus/integration/epoch_test.go

```diff
@@ -208,79 +193,85 @@ func TestEpochTransition_IdentitiesDisjoint(t *testing.T) {
 // to the next epoch upon reaching the appropriate view without any further changes
 // to the protocol state.
 func withNextEpoch(
```

Comment:
I have fixed this function a bunch of times already and the more I see changes to it the more confusing it gets :)
Not a point for this PR but I would say we should rethink how those tests are setup. Maybe we should create a root snapshot with next epoch immediately because what we do here is kind of hacky(in a way that we can't obtain such snapshot from root block, there must be at least some blocks to incorporate service events). 

If you agree we could create a low-priority issue to simplify test setup. Maybe instead of modifying root snapshot we just create new one similarly how we create root snapshot. Idk, need to think.